### PR TITLE
[testing] add cluster workers support

### PR DESCRIPTION
**Pull request by akichidis**
Follow up of the #492 .

This PR is giving a generous refactoring in the `Cluster` struct so we have proper organisation of the nodes. Now the cluster is spinning up Authority nodes. Each Authority has `one primary` and `multiple workers`. Now each node type (worker & primary) are responsible for start /stop and report running status - which makes the organisation much cleaner. Helper methods and input configuration has been provided to allow start / stop nodes in a flexible way. Also the transactions addresses of the workers can now be accessed in order to be able to send transactions to the cluster.

As before , the prometheus Registries for the primary & worker nodes can be accessed to observe the metrics.

Next steps: 
1. expose the receiver of the `tx_transaction_confirmation` consensus channel of the primary so tests can be written to monitor the consensus output transactions
2. enable starting the cluster with consensus off (help us test cases with external consensus)

Apart from the above an important next step is to create tests around the bootstrap sequence of nodes to start addressing [this](https://www.notion.so/mystenlabs/Narwhal-Storm-July-11-4f3cc6c4698448c5820d3e949bc3764e#b7b1cec7f6dc42df9fa6938935c9c506)

### CODE REVIEW

**Review by laura-makdah — APPROVED** [review]
If you find a way to expose `tr_transaction_confirmation` before merging this that would be 💯 otherwise I'll take it from here

File: test_utils/src/cluster.rs
Comment (huitseeker) [inline test_utils/src/cluster.rs:308]: This is probably a nit, I'll leave it to you to consider whether it's worth opening an issue to follow-up on this:

It is possible to me that the `ArcSwap` here might be an overkill: the `ArcSwap` is most useful to me when you have at least one mutation of a value from a non-empty state, i.e. `None` -> `Some(1)` -> `Some(42)`. (the `Option` is figurative here and just there to let me express the notion of "empty" state, not actually involved in this).

If I understand what you're trying to do correctly, you're more probably trying to initialize something from an empty state, and then bring it back to that empty state before setting it again, i.e. `None` -> `Some(1)`-> `None` -> `Some(42)`. For this consider that the `OnceCell` type has an initialization (and a `get_or_try_init` function) but also a `take` function that brings things back to this empty state.

**Review by huitseeker — APPROVED** [review]
Thanks a bunch!

File: primary/tests/causal_completion_tests.rs
Comment (akichidis) [reply primary/tests/causal_completion_tests.rs:65]: Ah I've added this to showcase @laura-makdah how could use the `tx_transaction_confirmation` to receive the consensus output

File: test_utils/src/cluster.rs
Comment (akichidis) [reply test_utils/src/cluster.rs:308]: Ah thanks for the above. To be honest my intuition was also that probably the `ArcSwap` might be an overkill. Unfortunately I can't use `OnceCell` as `PrimaryNodeDetails` needs to be cloned (`#[derive(Clone)]` is on top) which ends up in linter errors. I've switched to using an `Rc<RefCell>` instead as I believe is what I really need (interior mutability behind a way to share across clones)

File: consensus/src/consensus.rs
Comment (akichidis) [reply consensus/src/consensus.rs:95]: Good point, I'll amend this!

File: test_utils/src/cluster.rs
Comment (laura-makdah) [inline test_utils/src/cluster.rs:196]: tr isn't being used

**Pull request merged**
